packages/amplify-provider-awscloudformation/lib/attach-backend.js

```diff
@@ -0,0 +1,299 @@
+    await storeArtifactsForAmplifyService(context, awsConfig, DeploymentBucketName);
+  }
+
+  return currentAmplifyMeta;
+}
+
+async function storeArtifactsForAmplifyService(context, awsConfig, deploymentBucketName) {
+  const s3Client = getConfiguredS3Client(awsConfig);
+  const amplifyMetaFilePath = context.amplify.pathManager.getCurrentAmplifyMetaFilePath(process.cwd());
+  const backendConfigFilePath = context.amplify.pathManager.getCurrentBackendConfigFilePath(process.cwd());
+  await uploadFile(s3Client, deploymentBucketName, amplifyMetaFilePath);
+  await uploadFile(s3Client, deploymentBucketName, backendConfigFilePath);
+}
+
+async function uploadFile(s3Client, bucketName, filePath) {
+  if (!fs.existsSync(filePath)) {
+    return;
+  }
+  const key = path.basename(filePath);
+  const body = fs.createReadStream(filePath);
+  await s3Client
+    .putObject({
+      Bucket: bucketName,
+      Key: key,
+      Body: body,
+    })
+    .promise();
+}
+
+async function getAmplifyApp(context, amplifyClient) {
+  // If appId is in the inputParams, verify it
+  const { inputParams } = context.exeInfo;
+  if (inputParams.amplify && inputParams.amplify.appId) {
+    const inputAppId = inputParams.amplify.appId;
+    try {
+      const getAppResult = await amplifyClient
+        .getApp({
+          appId: inputAppId,
+        })
+        .promise();
+      context.print.info(`Input Amplify appId is verified: ${inputAppId}`);
+      context.print.info(`Amplify App name is: ${getAppResult.app.name}`);
+      return getAppResult.app;
+    } catch (e) {
+      context.print.error(`Input Amplify appId is invalild: ${inputAppId}`);
+      context.print.info(e);
+      throw e;
+    }
+  }
+
+  // If appId is not in the inputParams, prompt user to select
+  let apps = [];
+
+  let listAppsResponse = {};
+  do {
+    listAppsResponse = await amplifyClient
+      .listApps({
+        nextToken: listAppsResponse.nextToken,
+        maxResults: 25,
+      })
+      .promise();
+    apps = apps.concat(listAppsResponse.apps);
+  } while (listAppsResponse.nextToken);
+
+  if (apps.length > 1) {
+    const options = [];
+    apps.forEach(app => {
+      const option = {
+        name: `${app.name} / ${app.appId}`,
+        value: app,
+        short: app.appId,
+      };
+      options.push(option);
+    });
+
+    const { selection } = await inquirer.prompt({
+      type: 'list',
+      name: 'selection',
+      message: 'Which app are you working on?',
+      choices: options,
+    });
+
+    return selection;
+  } else if (apps.length === 1) {
+    context.print.info(`Found one amplify project '${apps[0].name} / ${apps[0].appId}'`);
+    const confirmMigrateMessage = 'Do you want to choose it?';
+    if (await context.prompt.confirm(confirmMigrateMessage)) {
+      return apps[0];
+    }
+  } else {
+    const errorMessage = `Found no currently existing amplify project.`;
```

Comment:
"No existing Amplify project was found"?

Reply:
wording id decided by the PM.

Reply:
No Amplify apps found. Please ensure your local profile matches the AWS account or region in which the Amplify app exists.

Reply:
changed